src/Aspire.Hosting/Eventing/DistributedApplicationEventing.cs

```diff
@@ -0,0 +1,41 @@
+namespace Aspire.Hosting.Eventing;
+
+internal class DistributedApplicationEventing : IDistributedApplicationEventing
+{
+    private readonly Dictionary<Type, List<DistributedApplicationEventSubscription>> _subscriptions = new();
+
+    public async Task PublishAsync<T>(T @event, CancellationToken cancellationToken) where T : IDistributedApplicationEvent
+    {
+        if (_subscriptions.TryGetValue(typeof(T), out var subscriptions))
+        {
+            foreach (var subscription in subscriptions)
```

Comment:
We probably need to worry about concurrency here. What happens if an event is subscribed/unsubscribed while the event is being fired? This is going to throw because the list is modified while enumerating it.

Reply:
Yep ... we will need to copy the list.